firedrake/function.py

```diff
@@ -543,6 +545,9 @@ def at(self, arg, *args, **kwargs):
         else:
             raise ValueError("Point dimension (%d) does not match geometric dimension (%d)." % (arg.shape[-1], gdim))
 
+        # Finish ensuring data is up-to-date for reading
+        self.dat.global_to_local_end(op2.READ)
```

Comment:
Why does this change happen?

Reply:
Apologies - this is not relevant to this PR. I've removed it in 71ffea43